src/snapred/backend/service/CalibrationService.py

```diff
@@ -157,7 +159,25 @@ def diffractionCalibration(self, request: DiffractionCalibrationRequest):
             convergenceThreshold=convergenceThreshold,
         )
 
-        return DiffractionCalibrationRecipe().executeRecipe(ingredients)
+        # get the needed input data
+        groceryList = [
```

Comment:
I dont think the consumer should have to specify the loader implementation, it cares not the means of getting the data, it just wants data.  

I would put this responsibility in the data layer, where it would pass back a wrapper containing workspace name and a grocerylistItem, then upon interaction with the wrapper, MantidSnapper would FetchGroceries using the parameters provided in the wrapper and return the workspace.

Reply:
The user doesn't **have** to specify the loader.  The default behavior is to call the mantid "Load" algorithm, which will figure it out.  Once mantid runs "Load", the correct loader will be stored here; maybe one day in the future, when all wars are over, we can also write SNAPRed to remember which loader was used and use it again.  
But further, the **dev** should know which loader to use, and this lets the dev specify the one to use.

Reply:
User =\/= Consumer. 
Consumer is more generalistic and in this case refers to code that consumes this recipe.

I am suggesting that the implementation details of data loading functionality shouldnt live in the service layer but the data layer.   If the datatypes or loaders change we'd have to change this in every service that used it, it should be abstracted behind an api so services dont have to bear that burden.  This breaks form from the architecture.

Reply:
Can I request we make this a phase 3 (or 4, or 5) fix, rather than a requirement right now?  Because that's going to take me at least a week to work out

Reply:
I'd rather not put it off too long, but it doenst have to be part of the pr.